src/Components/AdvancedSearch/Internal/QueryBuilder/QueryBuilder.styles.ts

```diff
@@ -102,35 +109,57 @@ export const getRowStyles = (
         firstColumn: [rowClassNames.firstColumn],
         inputColumn: [rowClassNames.inputColumn],
         buttonColumn: [rowClassNames.buttonColumn],
+        optionContainer: [
+            rowClassNames.optionContainer,
+            {
+                flex: '1 1 auto',
+                minWidth: 0,
+                textOverflow: 'ellipsis',
+                whiteSpace: 'nowrap',
+                overflow: 'hidden'
+            }
+        ],
         subComponentStyles: {
             deleteButton: {
                 rootDisabled: {
                     background: 'transparent'
                 }
+            },
+            propertyIcon: {
+                root: {
+                    paddingRight: 4,
```

Comment:
ideally we would just throw this in a Stack and let it take care of the spacing.

Reply:
Had to do some shenanigans to allow text overflow ellipsis, I can try with Stack but don't think it will be much cleaner.